src/IPM/utils.jl

```diff
@@ -38,20 +38,46 @@ has_constraints(solver) = solver.m != 0
 
 # Print functions -----------------------------------------------------------
 function print_init(solver::AbstractMadNLPSolver)
-    @notice(solver.logger,@sprintf("Number of nonzeros in constraint Jacobian............: %8i",get_nnzj(solver.nlp.meta)))
-    @notice(solver.logger,@sprintf("Number of nonzeros in Lagrangian Hessian.............: %8i\n",get_nnzh(solver.nlp.meta)))
+    @notice(solver.logger,@sprintf("Number of nonzeros in constraint Jacobian............: %8i", get_nnzj(solver.nlp.meta)))
+    @notice(solver.logger,@sprintf("Number of nonzeros in Lagrangian Hessian.............: %8i\n", get_nnzh(solver.nlp.meta)))
 
+    x_lb = get_lvar(solver.nlp)
+    x_ub = get_uvar(solver.nlp)
+    g_lb = get_lcon(solver.nlp)
+    g_ub = get_ucon(solver.nlp)
     num_fixed = length(solver.ind_fixed)
     num_var = get_nvar(solver.nlp) - num_fixed
     num_llb_vars = length(solver.ind_llb)
-    num_lu_vars = sum((get_lvar(solver.nlp).!=-Inf).*(get_uvar(solver.nlp).!=Inf)) - num_fixed
+    num_lu_vars = -num_fixed
+    # Number of bounded variables
+    for i in 1:get_nvar(solver.nlp)
+        if (x_lb[i] > -Inf) && (x_ub[i] < Inf)
+            num_lu_vars += 1
+        end
+    end
     num_uub_vars = length(solver.ind_uub)
-    num_eq_cons = sum(get_lcon(solver.nlp).==get_ucon(solver.nlp))
-    num_ineq_cons = sum(get_lcon(solver.nlp).!=get_ucon(solver.nlp))
-    num_ue_cons = sum((get_lcon(solver.nlp).!=get_ucon(solver.nlp)).*(get_lcon(solver.nlp).==-Inf).*(get_ucon(solver.nlp).!=Inf))
-    num_le_cons = sum((get_lcon(solver.nlp).!=get_ucon(solver.nlp)).*(get_lcon(solver.nlp).!=-Inf).*(get_ucon(solver.nlp).==Inf))
-    num_lu_cons = sum((get_lcon(solver.nlp).!=get_ucon(solver.nlp)).*(get_lcon(solver.nlp).!=-Inf).*(get_ucon(solver.nlp).!=Inf))
-    get_nvar(solver.nlp) < num_eq_cons && throw(NotEnoughDegreesOfFreedomException())
+    # Classify constraints
+    num_eq_cons, num_ineq_cons = 0, 0
+    num_ue_cons, num_le_cons, num_lu_cons = 0, 0, 0
+    for i in 1:get_ncon(solver.nlp)
```

Comment:
Can these for loops be replaced by functions? e.g., `num_eq_cons, num_ineq_cons, ... = get_cons_info(nlp)`